### PR TITLE
Add GetDoc support to Python completer

### PR DESCRIPTION
**Pull request by puremourning**
This builds on the Clang completer support and returns the docstring from any
    associated declarations for the context.

References: : 
* https://github.com/Valloric/YouCompleteMe/issues/1653
* https://github.com/Valloric/YouCompleteMe/pull/1694
* https://github.com/Valloric/ycmd/pull/229

### CODE REVIEW

File: ycmd/tests/subcommands_test.py [hunk 1]
```diff
@@ -1700,3 +1717,52 @@ def RunCompleterCommand_GetDocQuick_ClangCompleter_NoReadyToParse_test():
...
+                             contents = contents,
+                             command_arguments = [ 'GetDoc' ],
+                             completer_target = 'filetype_default' )
+
+  response = app.post_json( '/run_completer_command', event_data ).json
+
+  eq_( response, {
+       'detailed_info': '_ModuleMethod()\n\n'
+                      + 'Module method docs\n'
+                      + 'Are dedented, like you might expect',
```
Comment (micbou) [inline ycmd/tests/subcommands_test.py]: No `+` needed.

Comment (puremourning) [inline ycmd/tests/subcommands_test.py]: Fixed.

**Comment by puremourning** [comment]
Thanks for the review @micbou 

All fixed.

**Comment by micbou** [comment]
LGTM.

**Comment by Valloric** [comment]
Thanks!

@homu r+

**Pull request merged**